### ISSUE TITLE
[java] UnnecessaryLocalBeforeReturn false negative with annotated local

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#4168]
**Affects PMD Version:6.50.0** 

**Rule:UnnecessaryLocalBeforeReturn**

Please provide the rule name and a link to the rule documentation:
<https://pmd.sourceforge.io/pmd-6.50.0/pmd_rules_java_codestyle.html#unnecessarylocalbeforereturn>

**Description:**
Hi, I find a false negative about the rule UnnecessaryLocalBeforeReturn, please read the following code example. PMD should have reported a warning in line 3, but it does not. Hence, I think this is a false negative, and I find it is caused by annotation. Thanks.

**Code Sample demonstrating the issue:**

```java
import org.Annotation;

public class C {
    int[] getArr() {
        @Annotation int[] x = new int[] {1,2,3};  // should report a warning in this line
        return x;
    }
}
```

**Expected outcome:**

PMD should report a violation at line 3, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by adangel** [comment]
Thanks for the report.
However I think, this is not a false negative, because the local is annotated. In that case, you can't convert it into a single return statement, because you can't annotate the return statement...

That's a similar case when using suppresswarnings for cast before return:

```java
import java.util.List;

class Snippet {
    public static void main(String[] args) {
        Snippet s = new Snippet();
        System.out.println(s.getStringWithCastWarning());
        System.out.println(s.getStringSuppressedWarning());
    }

    @SuppressWarnings("rawtypes")
    private final List o = List.of("a");

    List<String> getStringWithCastWarning() {
        return (List<String>) o;
    }
    List<String> getStringSuppressedWarning() {
        @SuppressWarnings("unchecked")
        List<String> s = (List<String>) o;
        return s;
    }
    // List<String> getStringSuppressedWarningUncompilable() {
    //     @SuppressWarnings("unchecked")
    //     return (List<String>) o;
    // }
}
```
